In [48]:
%matplotlib inline

In [49]:
import pandas as pd
import matplotlib.pyplot as plt

In [23]:
#Read_the_data
train_df=pd.read_csv("Data/train.csv")
test_df=pd.read_csv("Data/test.csv")
df1=train_df.copy()

In [24]:
y_train=train_df['variety']

In [25]:
train_df = train_df.drop(columns="variety")

## Exploratory Data Analysis

In [26]:
train_df.head()

user_name    country  \
0            NaN  Australia   
1  @wawinereport         US   
2            NaN      Italy   
3     @vossroger     France   
4     @vossroger     France   

                                        review_title  \
0  Andrew Peace 2007 Peace Family Vineyard Chardo...   
1  North by Northwest 2014 Red (Columbia Valley (...   
2                  Renato Ratti 2007 Conca  (Barolo)   
3  Domaine l'Ancienne Cure 2010 L'Abbaye White (B...   
4  Château du Cèdre 2012 Le Cèdre Vintage Malbec ...   

                                  review_description            designation  \
0  Classic Chardonnay aromas of apple, pear and h...  Peace Family Vineyard   
1  This wine is near equal parts Syrah and Merlot...                    NaN   
2  Barolo Conca opens with inky dark concentratio...                  Conca   
3  It's impressive what a small addition of Sauvi...               L'Abbaye   
4  This ripe, sweet wine is rich and full of drie...       Le Cèdre Vintage   

   points  price          province                 region_1         region_2  \
0      83   10.0   Australia Other  South Eastern Australia              NaN   
1      89   15.0        Washington     Columbia Valley (WA)  Columbia Valley   
2      94   80.0          Piedmont                   Barolo              NaN   
3      87   22.0  Southwest France             Bergerac Sec              NaN   
4      88   33.0      France Other           Vin de Liqueur              NaN   

                    winery  
0             Andrew Peace  
1       North by Northwest  
2             Renato Ratti  
3  Domaine l'Ancienne Cure  
4         Château du Cèdre

In [27]:
train_df.describe()

points         price
count  82657.000000  77088.000000
mean      88.544564     36.922232
std        3.099649     43.698346
min       80.000000      4.000000
25%       86.000000     18.000000
50%       88.000000     27.000000
75%       91.000000     45.000000
max      100.000000   3300.000000

In [28]:
test_df.head()

user_name    country                                       review_title  \
0   @paulgwine          US  Boedecker Cellars 2011 Athena Pinot Noir (Will...   
1   @wineschach  Argentina  Mendoza Vineyards 2012 Gran Reserva by Richard...   
2       @vboone         US                Prime 2013 Chardonnay (Coombsville)   
3   @wineschach  Argentina  Bodega Cuarto Dominio 2012 Chento Vineyard Sel...   
4  @kerinokeefe      Italy           SassodiSole 2012  Brunello di Montalcino   

                                  review_description  \
0  Nicely differentiated from the companion Stewa...   
1  Charred, smoky, herbal aromas of blackberry tr...   
2  Slightly sour and funky in earth, this is a re...   
3  This concentrated, midnight-black Malbec deliv...   
4  Earthy aromas suggesting grilled porcini, leat...   

                      designation  points  price          province  \
0                          Athena      88   35.0            Oregon   
1  Gran Reserva by Richard Bonvin      90   60.0  Mendoza Province   
2                             NaN      87   38.0        California   
3       Chento Vineyard Selection      91   20.0  Mendoza Province   
4                             NaN      90   49.0           Tuscany   

                 region_1           region_2                 winery  
0       Willamette Valley  Willamette Valley      Boedecker Cellars  
1                 Mendoza                NaN      Mendoza Vineyards  
2             Coombsville               Napa                  Prime  
3                 Mendoza                NaN  Bodega Cuarto Dominio  
4  Brunello di Montalcino                NaN            SassodiSole

In [29]:
# Read about the Y data

In [35]:
from collections import Counter

In [41]:
c=Counter(y_train)

In [44]:
y_train.describe()

count          82657
unique            28
top       Pinot Noir
freq           10587
Name: variety, dtype: object

In [46]:
c.most_common(5)

[('Pinot Noir', 10587),
 ('Chardonnay', 9403),
 ('Cabernet Sauvignon', 7552),
 ('Red Blend', 7166),
 ('Bordeaux-style Red Blend', 5497)]

## Top 5 varieties of wines with their counts

- ('Pinot Noir', 10587),
- ('Chardonnay', 9403),
- ('Cabernet Sauvignon', 7552),
- ('Red Blend', 7166),
- ('Bordeaux-style Red Blend', 5497)

In [ ]:
plt.hist(y_train)